spec/javascripts/govuk_publishing_components/lib/auto-scroll-tracker-spec.js

```diff
@@ -0,0 +1,234 @@
+  var scrollTracker, scrollTracker2
+
+  beforeEach(function () {
+    window.GOVUK.setCookie('cookies_policy', '{"essential":true,"settings":true,"usage":true,"campaigns":true}')
+    jasmine.clock().install()
+    spyOn(GOVUK.analytics, 'trackEvent')
+  })
+
+  afterEach(function () {
+    stopComponent(scrollTracker)
+    if (scrollTracker2) {
+      stopComponent(scrollTracker2)
+    }
+    jasmine.clock().uninstall()
+
+    window.GOVUK.deleteCookie('cookies_policy')
+    window.GOVUK.analyticsVars.scrollTrackerStarted = false
+
+    if (GOVUK.analytics.trackEvent.calls) {
+      GOVUK.analytics.trackEvent.calls.reset()
+    }
+  })
+
+  // remove all the event listeners, otherwise conflicts occur
+  function stopComponent (tracker) {
+    window.removeEventListener('scroll', tracker.scrollEvent)
+    window.removeEventListener('resize', tracker.resizeEvent)
+    clearInterval(tracker.interval)
+  }
+
+  it('should only initialise once on a page', function () {
+    var el = document.createElement('div')
+    scrollTracker = new GOVUK.Modules.AutoScrollTracker(el)
+    spyOn(scrollTracker, 'getWindowDetails')
+    scrollTracker.init()
+
+    var el2 = document.createElement('div')
+    scrollTracker2 = new GOVUK.Modules.AutoScrollTracker(el2)
+    spyOn(scrollTracker2, 'getWindowDetails')
+    scrollTracker2.init()
+
+    expect(scrollTracker.getWindowDetails).toHaveBeenCalled()
+    expect(scrollTracker2.getWindowDetails).not.toHaveBeenCalled()
+  })
+
+  describe('when tracking headings', function () {
+    var el
+
+    beforeEach(function () {
+      var h3pos = window.innerHeight + 1000
+      var FIXTURE =
+        '<div style="position: absolute; top: 0px;">' +
+          '<h1>Heading 1</h1>' +
+          '<div id="wrapper">' +
+            '<h2 style="display: none;">Heading 2</h2>' +
+            '<h3 style="margin-top: ' + h3pos + 'px;">Heading 3</h3>' +
+          '</div>' +
+        '</div>'
+      el = document.createElement('div')
+      el.setAttribute('data-track-type', 'headings')
+      el.innerHTML = FIXTURE
+      document.body.appendChild(el)
+      scrollTracker = new GOVUK.Modules.AutoScrollTracker(el)
+      scrollTracker.init()
+    })
+
+    afterEach(function () {
+      document.body.removeChild(el)
+    })
+
+    it('should send a tracking event on page load for headings that are already visible', function () {
+      expect(GOVUK.analytics.trackEvent.calls.count()).toBe(1)
+      expect(GOVUK.analytics.trackEvent.calls.argsFor(0)).toEqual(['ScrollTo', 'Heading', { label: 'Heading 1', nonInteraction: true }])
+    })
+
+    it('should not track a heading more than once', function () {
+      GOVUK.analytics.trackEvent.calls.reset()
+      GOVUK.triggerEvent(document.body, 'resize')
+      jasmine.clock().tick(200)
+
+      expect(GOVUK.analytics.trackEvent.calls.count()).toBe(0)
+    })
+
+    it('should track headings on scroll and ignore already tracked headings', function () {
```

Comment:
I found this test and the next couple a little confusing by the terminology of "on scroll" when the test isn't triggering a scroll event. I'd imagine just removing "on scroll" from the test description would clear that up (I imagining you're using resize as that doesn't carry the check that scroll does that the user has changed scroll position)

Reply:
I've tweaked the test a little, to keep the scroll event. Good point.